src/dsp_tools/commands/excel2json/input_error.py

```diff
@@ -126,3 +126,39 @@ def execute_error_protocol(self) -> str:
         if self.message_path:
             msg.append(f"The error occurred at {self.message_path}")
         return separator.join(msg)
+
+
+@dataclass(frozen=True)
+class JsonValidationResourceProblem:
+    """This class contains information about a JSON resource section that fails its validation against the schema."""
+
+    problematic_resource: str | None = None
+    excel_sheet: str | None = None
+    excel_column: str | None = None
+    excel_row: int | None = None
+    original_msg: str | None = None
+    message_path: str | None = None
+
+    def execute_error_protocol(self) -> str:
+        """
+        This function initiates all the steps for successful problem communication with the user.
+
+        Returns:
+            message for the error
+        """
+        msg = [
+            f"{separator}Section of the problem: 'Resources'",
+        ]
+        if self.problematic_resource:
+            msg.append(f"Problematic Resource '{self.problematic_resource}'")
+        if self.excel_sheet:
+            msg.append(f"The problem is caused by the value in the Excel sheet '{self.excel_sheet}'")
+        if self.excel_row:
+            msg.append(f"The problem is caused by the value in the Excel row {self.excel_row}")
+        if self.excel_column:
+            msg.append(f"The problem is caused by the value in the Excel column '{self.excel_column}'")
```

Comment:
if my assumption is correct (see above) that these are all defined in the same cases, then they could be bundled:

```suggestion
        if self.excel_sheet and self.excel_row and self.excel_column:
            msg.append(f"The problem is caused by the value in the Excel sheet '{self.excel_sheet}' at row {self.excel_row}, column {self.excel_column}.")
```